pkg/server/singleprocess/service_deploy.go

```diff
@@ -127,7 +128,7 @@ func (s *Service) GetDeployment(
 
 	d, err := s.state(ctx).DeploymentGet(req.Ref)
 	if err != nil {
-		return nil, err
+		return nil, hcerr.Externalize(hclog.FromContext(ctx), err, "failed to get deployment", "id", req.Ref.Target)
```

Comment:
`req.Ref.Target` is a bit of a weird one. It's possible to get deployments either by ID or by sequence number, so Target can be either of those: https://github.com/hashicorp/waypoint/blob/a0d9c3d198617858909e816233cb367d7652883a/pkg/server/proto/server.proto#L942-L947

That said, I gave this a test, and the logger actually formats it ok!

```
$ waypoint deploy

» Deploying web...
  Performing operation locally
✓ Running deploy v1
✓ Setting up network...
✓ Starting container
✓ App deployed as container: web-01GA3SSGPMANZ1TQ7662R1HTDM
✓ Docker image "waypoint.local/web:latest" up to date!
! failed to get deployment
  "id": "&{01GA3SSGA608XDMQ0KQWCJRYN2}"
```

It's not beautiful, but it's better than omitting the context I think. The one change i'd make:

```suggestion
		return nil, hcerr.Externalize(hclog.FromContext(ctx), err, "failed to get deployment", "target", req.Ref.Target)
```